dashboard-all-the-things/src/app/dashboard-carousel/dashboard-carousel.component.ts

```diff
@@ -1,51 +1,64 @@
-import { Component, OnInit, OnDestroy, ElementRef } from '@angular/core';
-import { Subscription } from 'rxjs';
-import { Observable } from 'rxjs/Observable';
+import { ExternalUrlComponent } from '../dashboards/external-url/external-url.component';
+import { ReviewComponent } from '../dashboards/review';
+import { SonarCoverageComponent } from '../dashboards/sonar-coverage';
+import { StackOverflowComponent } from '../dashboards/stackoverflow/stackoverflow.component';
+import { WidgetEvent } from '../dashboards/WidgetEvent';
+import { DynamicComponent } from './dynamic.component';
+import { SocketService } from './socket.service';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ConnectableObservable, Subscription } from 'rxjs';
+
+let COMPONENTS = [
+  { key: "coverage", type: SonarCoverageComponent },
+  { key: "ciWall", type: ExternalUrlComponent },
+  { key: "reviews", type: ReviewComponent },
+  { key: "stackoverflow", type: StackOverflowComponent }
+];
 
 @Component({
   selector: 'app-dashboard-carousel',
   templateUrl: 'dashboard-carousel.component.html',
-  styleUrls: ['dashboard-carousel.component.scss']
+  styleUrls: ['dashboard-carousel.component.scss'],
+  entryComponents: COMPONENTS.map((el) => el.type)
+    .filter((elem, index, arr) => arr.indexOf(elem) === index)
```

Comment:
@takecare the scope of that function is so small that having a longer name wouldn't really add any value, but a bit of noise, we'll use the same for both  tho 👍